File: packages/ios/Sources/Components/Menu/Menu.figma.swift
import SwiftUI
import Figma

struct KozmosMenuBasicConnect: FigmaConnect {
    let component = KozmosMenu.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=101-8170"
    var variant = ["Content": "Basic"]

    @FigmaString("Label Text")
    var label: String = "Actions"

    @FigmaString("Item 1 Text")
    var item1: String = "Rename"

    @FigmaString("Item 2 Text")
    var item2: String = "Duplicate"

    @FigmaString("Item 3 Text")
    var item3: String = "Archive"

    @FigmaString("Shortcut Text")
    var shortcut: String = "R"

    var body: some View {
        KozmosMenu(
            title: "Open menu",
            content: KozmosMenuContent(
                label: self.label,
                items: [
                    KozmosMenuItem(text: self.item1, shortcut: self.shortcut),
                    KozmosMenuItem(text: self.item2),
                    KozmosMenuItem(text: self.item3)
                ],
                contentType: .basic
            )
        )
    }
}

struct KozmosMenuCheckboxConnect: FigmaConnect {
    let component = KozmosMenu.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=101-8170"
    var variant = ["Content": "Checkbox"]

    @FigmaString("Label Text")
    var label: String = "Map layers"

    @FigmaString("Item 1 Text")
    var item1: String = "Transit"

    @FigmaString("Item 2 Text")
    var item2: String = "Amenities"

    @FigmaString("Item 3 Text")
    var item3: String = "Accessibility"

    var body: some View {
        KozmosMenu(
            title: "Map layers",
            content: KozmosMenuContent(
                label: self.label,
                items: [
                    KozmosMenuItem(text: self.item1, isChecked: true),
                    KozmosMenuItem(text: self.item2),
                    KozmosMenuItem(text: self.item3, isChecked: true)
                ],
                contentType: .checkbox
            )
        )
    }
}

struct KozmosMenuRadioConnect: FigmaConnect {
    let component = KozmosMenu.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=101-8170"
    var variant = ["Content": "Radio"]

    @FigmaString("Label Text")
    var label: String = "Route preference"

    @FigmaString("Item 1 Text")
    var item1: String = "Fastest"

    @FigmaString("Item 2 Text")
    var item2: String = "Accessible"

    @FigmaString("Item 3 Text")
    var item3: String = "Fewest transfers"

    var body: some View {
        KozmosMenu(
            title: "Route preference",
            content: KozmosMenuContent(
                label: self.label,
                items: [
                    KozmosMenuItem(text: self.item1, isSelected: true),
                    KozmosMenuItem(text: self.item2),
                    KozmosMenuItem(text: self.item3)
                ],
                contentType: .radio
            )
        )
    }
}

struct KozmosMenuSubmenuConnect: FigmaConnect {
    let component = KozmosMenu.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=101-8170"
    var variant = ["Content": "Submenu"]

    @FigmaString("Label Text")
    var label: String = "Workspace"

    @FigmaString("Item 1 Text")
    var item1: String = "Members"

    @FigmaString("Item 2 Text")
    var item2: String = "Settings"

    @FigmaString("Item 3 Text")
    var item3: String = "Share"

    var body: some View {
        KozmosMenu(
            title: "Workspace",
            content: KozmosMenuContent(
                label: self.label,
                items: [
                    KozmosMenuItem(text: self.item1),
                    KozmosMenuItem(text: self.item2),
                    KozmosMenuItem(
                        text: self.item3,
                        submenuItems: ["Invite people", "Export"]
                    )
                ],
                contentType: .submenu
            )
        )
    }
}

File: packages/android/src/main/java/com/kozmos/components/Menu/Menu.figma.kt
package com.kozmos.components.menu

import androidx.compose.runtime.Composable
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType
import com.figma.code.connect.FigmaVariant

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=101-8170")
@FigmaVariant("Content", "Basic")
class KozmosMenuBasicConnect {
    @FigmaProperty(FigmaType.Text, "Label Text")
    val label: String = "Actions"

    @FigmaProperty(FigmaType.Text, "Item 1 Text")
    val item1: String = "Rename"

    @FigmaProperty(FigmaType.Text, "Item 2 Text")
    val item2: String = "Duplicate"

    @FigmaProperty(FigmaType.Text, "Item 3 Text")
    val item3: String = "Archive"

    @FigmaProperty(FigmaType.Text, "Shortcut Text")
    val shortcut: String = "R"

    @Composable
    fun ComponentExample() {
        KozmosMenu(
            expanded = true,
            onDismissRequest = {},
            content = KozmosMenuContent(
                label = label,
                items = listOf(
                    KozmosMenuItem(text = item1, shortcut = shortcut),
                    KozmosMenuItem(text = item2),
                    KozmosMenuItem(text = item3)
                ),
                contentType = KozmosMenuContentType.Basic
            )
        )
    }
}

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=101-8170")
@FigmaVariant("Content", "Checkbox")
class KozmosMenuCheckboxConnect {
    @FigmaProperty(FigmaType.Text, "Label Text")
    val label: String = "Map layers"

    @FigmaProperty(FigmaType.Text, "Item 1 Text")
    val item1: String = "Transit"

    @FigmaProperty(FigmaType.Text, "Item 2 Text")
    val item2: String = "Amenities"

    @FigmaProperty(FigmaType.Text, "Item 3 Text")
    val item3: String = "Accessibility"

    @Composable
    fun ComponentExample() {
        KozmosMenu(
            expanded = true,
            onDismissRequest = {},
            content = KozmosMenuContent(
                label = label,
                items = listOf(
                    KozmosMenuItem(text = item1, checked = true),
                    KozmosMenuItem(text = item2),
                    KozmosMenuItem(text = item3, checked = true)
                ),
                contentType = KozmosMenuContentType.Checkbox
            )
        )
    }
}

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=101-8170")
@FigmaVariant("Content", "Radio")
class KozmosMenuRadioConnect {
    @FigmaProperty(FigmaType.Text, "Label Text")
    val label: String = "Route preference"

    @FigmaProperty(FigmaType.Text, "Item 1 Text")
    val item1: String = "Fastest"

    @FigmaProperty(FigmaType.Text, "Item 2 Text")
    val item2: String = "Accessible"

    @FigmaProperty(FigmaType.Text, "Item 3 Text")
    val item3: String = "Fewest transfers"

    @Composable
    fun ComponentExample() {
        KozmosMenu(
            expanded = true,
            onDismissRequest = {},
            content = KozmosMenuContent(
                label = label,
                items = listOf(
                    KozmosMenuItem(text = item1, selected = true),
                    KozmosMenuItem(text = item2),
                    KozmosMenuItem(text = item3)
                ),
                contentType = KozmosMenuContentType.Radio
            )
        )
    }
}

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=101-8170")
@FigmaVariant("Content", "Submenu")
class KozmosMenuSubmenuConnect {
    @FigmaProperty(FigmaType.Text, "Label Text")
    val label: String = "Workspace"

    @FigmaProperty(FigmaType.Text, "Item 1 Text")
    val item1: String = "Members"

    @FigmaProperty(FigmaType.Text, "Item 2 Text")
    val item2: String = "Settings"

    @FigmaProperty(FigmaType.Text, "Item 3 Text")
    val item3: String = "Share"

    @Composable
    fun ComponentExample() {
        KozmosMenu(
            expanded = true,
            onDismissRequest = {},
            content = KozmosMenuContent(
                label = label,
                items = listOf(
                    KozmosMenuItem(text = item1),
                    KozmosMenuItem(text = item2),
                    KozmosMenuItem(
                        text = item3,
                        submenuItems = listOf("Invite people", "Export")
                    )
                ),
                contentType = KozmosMenuContentType.Submenu
            )
        )
    }
}
